### ISSUE TITLE
What's the meaning of shadingPoolIDs_t?

### ISSUE THREAD

**Issue by Michael-Lfx** [Think-Silicon/GLOVE#37]
**What's the meaning of the member `shadingPoolIDs_t mShadingObjectPool` of `ResourceManager`?**
Both `Context::glCreateShader` and `Context::CreateProgram` call `mResourceManager->GetShader` then store `(type, id)` to `mShadingObjectPool`. 
**Why not just do it in `GetObject` of `ObjectArray<T>`?** With *object id* inside those model class, this could simplify members of `ActiveObjects` if I am not getting it wrong.

Any help is appreciated.

https://github.com/Think-Silicon/GLOVE/blob/71366b1b1b33729a3b725640c0d0f86d801d69fc/GLES/source/context/contextShader.cpp#L59
https://github.com/Think-Silicon/GLOVE/blob/71366b1b1b33729a3b725640c0d0f86d801d69fc/GLES/source/context/contextShaderProgram.cpp#L92
https://github.com/Think-Silicon/GLOVE/blob/71366b1b1b33729a3b725640c0d0f86d801d69fc/GLES/source/resources/resourceManager.h#L63

**Comment by abasilak** [comment]
Yes you are correct. Ideally, we want to split _mShadingObjectPool_ into to two _ObjectArray<T> objects_.
